### PR TITLE
Ipv6 template

### PR DESCRIPTION
**Pull request by EricTheRed999**
This is my first PR for a sample script...so please review carefully.   

IMPORTANT:   Please do NOT publish before Tue Apr23 as this feature is not yet released for public preview.

### CODE REVIEW

**Comment by EricTheRed999** [comment]
To Reviewers:   Please Help with these validation failures:   
1.	Still doesn’t like my [valid] Git username – EricTheRed999
2.	The validator seems to be trying to do a deployment of this feature that is still in private preview.  Is this same Subscription ID used for all validator deployments?   If so, I can add the preview feature flag.  

\"code\": \"SubscriptionNotRegisteredForFeature\",\r\n    \"message\": \"Subscription f42b4319-067a-4548-a650-33a1553b3a42 is not registered for feature Microsoft.Network/AllowIPv6VirtualNetwork required to carry out the requested operation.\

**Comment by EricTheRed999** [comment]
IMPORTANT:   Reviewers, please IGNORE my previous request to hold publication of this script until Apr23...we'd like you to publish this script as soon as possible so we can link to the documents which will go live on Apr23.

File: ipv6-in-vnet/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+   },
+  "variables": {
+    "vnetv4AddressRange": "10.0.0.0/16",
+    "vnetv6AddressRange": "ace:cab:deca::/48",
+    "subnetv4AddressRange": "10.0.0.0/24",
+    "subnetv6AddressRange": "ace:cab:deca:deed::/64",
+    "subnetName": "DualStackSubnet",
+    "newStorageAccountName": "[concat('st', uniqueString(resourceGroup().id))]",
+    "availabilitySetName": "myavset",
+    "subnet-id": "[concat(resourceId('Microsoft.Network/virtualNetworks','VNET'),'/subnets/',variables('subnetName'))]",
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: remove concat, just use resourceId

resourceId('Microsoft.Network/virtualNetworks/subnets', 'VNET', variables('subnetName'))

File: ipv6-in-vnet/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+    "ApiVersion": "2018-07-01",  
+    "numberOfInstances": 2,
+    "vmName": "DsVM",
+    "publicipName": "RDPpublicIp",
+    "imagePublisher": "MicrosoftWindowsServer",
+    "imageOffer": "WindowsServer",
+    "imageSku": "2016-Datacenter"
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: any reason we can't use 2019?

File: ipv6-in-vnet/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+  },
+  "resources": [
+    {
+      "apiVersion": "[variables('ApiVersion')]",
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: Use a literal apiVersion so schema validation can be done.

File: ipv6-in-vnet/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+      "type": "Microsoft.Network/publicIPAddresses",
+      "name": "[concat(variables('publicipName'), copyindex())]",
+      "copy": {
+        "name": "publicIpLoop",
+        "count": "[variables('numberOfInstances')]"
+      },
+      "location": "[resourceGroup().location]",
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: ipv6-in-vnet/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+      "properties": {
+        "publicIPAllocationMethod": "Dynamic"
+      }
+    },
+	{
+      "apiVersion": "[variables('ApiVersion')]",
+      "type": "Microsoft.Network/publicIPAddresses",
+      "name": "lbpublicip-v6",
+      "location": "[resourceGroup().location]",
+      "properties": {
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: format the JSON, e.g. SHIFT+ALT+F in VS Code

File: ipv6-in-vnet/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+        "publicIPAllocationMethod": "Dynamic",
+		"publicIPAddressVersion": "IPv6"
+      }
+    },
+    {
+      "type": "Microsoft.Compute/availabilitySets",
+      "name": "[variables('availabilitySetName')]",
+      "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: update apiVersion

File: ipv6-in-vnet/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "platformFaultDomainCount": "1",
+        "platformUpdateDomainCount": "1",
+        "managed": "true"
+      }
+    },
+    {
+      "apiVersion": "2015-06-15",
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: update apiVersion

File: ipv6-in-vnet/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,370 @@
...
+		"publicIPAddressVersion": "IPv6"
+      }
+    },
+    {
+      "type": "Microsoft.Compute/availabilitySets",
+      "name": "[variables('availabilitySetName')]",
+      "apiVersion": "2016-04-30-preview",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "platformFaultDomainCount": "1",
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: use 2 and 5 here

File: ipv6-in-vnet/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,370 @@
...
+        "platformUpdateDomainCount": "1",
+        "managed": "true"
+      }
+    },
+    {
+      "apiVersion": "2015-06-15",
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('newStorageAccountName')]",
+      "location": "eastus2",
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: ipv6-in-vnet/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,370 @@
...
+          },
+		  {
+            "name": "LBBAP-v6"
+          }
+        ],
+        "loadBalancingRules": [
+          {
+            "properties": {
+              "frontendIPConfiguration": {
+                "id": "[concat(resourceId('Microsoft.Network/loadBalancers', 'loadBalancer'), '/frontendIpConfigurations/LBFE')]"
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: remove concat and use resourceId (here and below)

resourceId('Microsoft.Network/loadBalancers/frontendIpConfigurations', 'loadBalancer', 'LBFE')

File: ipv6-in-vnet/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,370 @@
...
+      "copy": {
+        "name": "netIntLoop",
+        "count": "[variables('numberOfInstances')]"
+      },
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "Microsoft.Network/virtualNetworks/VNET",
+        "Microsoft.Network/networkSecurityGroups/dsNsg",
+        "Microsoft.Network/loadBalancers/loadBalancer",
+		"[concat('Microsoft.Network/publicIPAddresses/',variables('publicipName'),copyindex())]"
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: use "publicIpLoop"

File: ipv6-in-vnet/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,370 @@
...
+      "name": "[concat(variables('vmName'), copyindex())]",
+      "copy": {
+        "name": "virtualMachineLoop",
+        "count": "[variables('numberOfInstances')]"
+      },
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[resourceId('Microsoft.Storage/storageAccounts',variables('newStorageAccountName'))]",
+        "[concat('Microsoft.Network/networkInterfaces/',variables('vmName'),copyindex())]",
+        "[concat('Microsoft.Compute/availabilitySets/', variables('availabilitySetName'))]"
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: use resourceId

File: ipv6-in-vnet/azuredeploy.json [hunk 13]
```diff
@@ -0,0 +1,370 @@
...
+      "type": "Microsoft.Compute/virtualMachines",
+      "name": "[concat(variables('vmName'), copyindex())]",
+      "copy": {
+        "name": "virtualMachineLoop",
+        "count": "[variables('numberOfInstances')]"
+      },
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[resourceId('Microsoft.Storage/storageAccounts',variables('newStorageAccountName'))]",
+        "[concat('Microsoft.Network/networkInterfaces/',variables('vmName'),copyindex())]",
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: use netIntLoop

File: ipv6-in-vnet/azuredeploy.json [hunk 14]
```diff
@@ -0,0 +1,370 @@
...
+                  "id": "[concat(resourceId('Microsoft.Network/loadBalancers','loadBalancer'),'/backendAddressPools/LBBAP-v6')]"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    },
+    {
+      "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: update apiVersion

File: ipv6-in-vnet/azuredeploy.json [hunk 15]
```diff
@@ -0,0 +1,370 @@
...
+              "primary": "true",
+              "publicIPAddress": {
+                "id": "[resourceId('Microsoft.Network/publicIPAddresses',concat(variables('publicipName'),copyindex()))]"
+              },
+              "subnet": {
+                "id": "[variables('subnet-id')]"
+              },
+              "loadBalancerBackendAddressPools": [
+                {
+                  "id": "[concat(resourceId('Microsoft.Network/loadBalancers','loadBalancer'),'/backendAddressPools/LBBAP')]"
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.json]: use resourceId here and below

File: ipv6-in-vnet/azuredeploy.parameters.json [hunk 16]
```diff
@@ -0,0 +1,12 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminUsername": {
+       "value": "GEN-USER"
```
Comment (bmoore-msft) [inline ipv6-in-vnet/azuredeploy.parameters.json]: GEN-UNIQUE

**Pull request merged**